internal/shim-kvm/.cargo/config

```diff
@@ -1,10 +1,14 @@
 [build]
-target = "x86_64-unknown-linux-musl"
+# FIXME: "x86_64-unknown-none" is not yet a tier 2 target, so setting this as
+#        the default target breaks rust-analyzer.
+# target = "x86_64-unknown-none"
 
-[target.x86_64-unknown-linux-musl]
+[target.x86_64-unknown-none]
 rustflags = [
     "-C", "relocation-model=pic",
+    "-C", "linker=gcc",
```

Comment:
@bstrie  has this working on Friday with the default linker. It only required two small linker script changes.

Reply:
something like this?
```
diff --git a/internal/shim-kvm/layout.ld b/internal/shim-kvm/layout.ld
index 9a7e8ff0..8b79e947 100644
--- a/internal/shim-kvm/layout.ld
+++ b/internal/shim-kvm/layout.ld
@@ -51,14 +51,12 @@ SECTIONS {
         QUAD(4 * 0x40000000 + 0x83); /* Flags::HUGE_PAGE | Flags::WRITABLE | Flags::PRESENT */
         QUAD(5 * 0x40000000 + 0x83); /* Flags::HUGE_PAGE | Flags::WRITABLE | Flags::PRESENT */
         FILL(0);
-        . = CONSTANT(COMMONPAGESIZE);
     } :pagetables
 
     . = _ENARX_PML4;
     .pml4 : ALIGN(CONSTANT(COMMONPAGESIZE)) {
         QUAD(_ENARX_PML3 + 0x3); /* Flags::WRITABLE | Flags::PRESENT */
         FILL(0);
-        . = CONSTANT(COMMONPAGESIZE);
     } :pagetables
 
     . = _ENARX_CPUID;
```

Reply:
> something like this?
> 
> ```diff
> diff --git a/internal/shim-kvm/layout.ld b/internal/shim-kvm/layout.ld
> index 9a7e8ff0..8b79e947 100644
> --- a/internal/shim-kvm/layout.ld
> +++ b/internal/shim-kvm/layout.ld
> @@ -51,14 +51,12 @@ SECTIONS {
>          QUAD(4 * 0x40000000 + 0x83); /* Flags::HUGE_PAGE | Flags::WRITABLE | Flags::PRESENT */
>          QUAD(5 * 0x40000000 + 0x83); /* Flags::HUGE_PAGE | Flags::WRITABLE | Flags::PRESENT */
>          FILL(0);
> -        . = CONSTANT(COMMONPAGESIZE);
>      } :pagetables
>  
>      . = _ENARX_PML4;
>      .pml4 : ALIGN(CONSTANT(COMMONPAGESIZE)) {
>          QUAD(_ENARX_PML3 + 0x3); /* Flags::WRITABLE | Flags::PRESENT */
>          FILL(0);
> -        . = CONSTANT(COMMONPAGESIZE);
>      } :pagetables
>  
>      . = _ENARX_CPUID;
> ```

That's not quiet enough: With only those changes `_ENARX_MEM_START`, `_ENARX_SHIM_START`, etc are not associated with a section so they won't get relocated:
```bash
readelf -a ./target/debug/build/enarx-c6bb1d3dd45a0b0e/out/internal/shim-kvm/x86_64-unknown-none/debug/shim-kvm | grep _ENARX
  1682: 00000000ffe00000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_MEM_START
  1689: 0000000100187000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_EXEC_START
  1697: 0000000108187000     0 NOTYPE  GLOBAL DEFAULT   22 _ENARX_EXEC_END
  3103: 00000000ffe02000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_CPUID
  4646: 00000000ffe04000     0 NOTYPE  GLOBAL DEFAULT    5 _ENARX_SECRETS
  5753: 00000000fffff000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_SHIM_START
  5754: 00000000ffe00000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_PML3
  5755: 00000000ffe01000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_PML4
  5756: 0000000008000000     0 NOTYPE  GLOBAL DEFAULT  ABS _ENARX_EXEC_LEN
  5757: 00000000ffe03000     0 NOTYPE  GLOBAL DEFAULT    4 _ENARX_GHCB
```
This eventually causes this code to fail because it assumes that `_ENARX_MEM` is pointing to the relocated address:
https://github.com/enarx/enarx/blob/ed4cf47999fcacbf888dc0156f53d9e7cfebdf99/internal/shim-kvm/src/allocator.rs#L149-L150.